src/store/swapOrderSlice.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * Listen for changes in the swap order status and determines if a notification should be shown
+ *
+ * Some gotchas:
+ * If the status of an order is created, presignaturePending, open - we always display a notification.
+ * Here it doesn't matter if the order was started through the UI or the gateway returned that order on a new browser instance.
+ *
+ * For fulfilled, expired, cancelled - we only display a notification if the old status is not undefined.
+ * Why? Because if the status is undefined, it means that the order was just fetched from the gateway, and
+ * it was already processed and there is no need to show a notification. If the status is != undefined, it means
+ * that the user has started the swap through the UI (or has continued it from a previous state), and we should show a notification.
+ *
+ * @param listenerMiddleware
+ */
+export const swapOrderStatusListener = (listenerMiddleware: typeof listenerMiddlewareInstance) => {
+  listenerMiddleware.startListening({
+    actionCreator: slice.actions.setSwapOrder,
+    effect: (action, listenerApi) => {
+      const { dispatch } = listenerApi
+      const swapOrder = action.payload
+      const oldStatus = selectSwapOrderStatus(listenerApi.getOriginalState(), swapOrder.orderUid)
+      const newStatus = swapOrder.status
+
+      if (oldStatus === newStatus || newStatus === undefined) {
+        return
+      }
+
+      switch (newStatus) {
+        case 'created':
+          const safeInfo = selectSafeInfo(listenerApi.getState())
+
+          dispatch(
+            showNotification({
+              title: 'Order created',
+              message:
+                safeInfo.data?.threshold === 1
+                  ? 'Waiting for the transaction to be executed'
+                  : 'Waiting for confirmation from signers of your Safe',
+              groupKey,
+              variant: 'info',
+            }),
+          )
+
+          break
+        case 'presignaturePending':
+          dispatch(
+            showNotification({
+              title: 'Order waiting for signature',
+              message: 'Waiting for confirmation from signers of your Safe',
+              groupKey,
+              variant: 'info',
+            }),
+          )
+          break
+        case 'open':
+          dispatch(
+            showNotification({
+              title: 'Order transaction confirmed',
+              message: 'Waiting for order execution by the CoW Protocol',
+              groupKey,
+              variant: 'info',
+            }),
+          )
+          break
+        case 'fulfilled':
+          if (oldStatus === undefined) {
+            return
+          }
+          dispatch(
+            showNotification({
+              title: 'Order executed',
+              message: 'Your order has been successful',
+              groupKey,
+              variant: 'success',
+            }),
+          )
+          dispatch(slice.actions.deleteSwapOrder(swapOrder.orderUid))
```

Comment:
I think we need to dispatch the deletion as the first thing in case `oldStatus === undefined` we wouldn't delete the entry. This can happen when a swap gets filled and the user navigates back to the swap page. The widget emits a fulfilled event so we dispatch the state again but never remove it after that.